airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/shared/ActivityConfiguration.java

```diff
@@ -24,12 +26,20 @@
   private static final int MAX_SYNC_TIMEOUT_DAYS = configs.getSyncJobMaxTimeoutDays();
   private static final Duration DB_INTERACTION_TIMEOUT = Duration.ofSeconds(configs.getMaxActivityTimeoutSecond());
 
+  // retry infinitely if the worker is killed without exceptions and dies due to timeouts
+  // but fail for everything thrown by the call itself which is rethrown as runtime exceptions
+  private static final RetryOptions ORCHESTRATOR_RETRY = RetryOptions.newBuilder()
+      .setDoNotRetry(RuntimeException.class.getName(), WorkerException.class.getName())
```

Comment:
TODO: double check that the tests won't pass if this is set to NO_RETRY

Reply:
TODO: add tests to verify that an NPE, workerexception, etc are actually not retried here.